pallets/maintenance-mode/src/lib.rs

```diff
@@ -0,0 +1,147 @@
+	use frame_support::pallet_prelude::*;
+	use frame_support::traits::{EnsureOrigin, Filter};
+	use frame_system::pallet_prelude::*;
+
+	/// Pallet for migrations
+	#[pallet::pallet]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	/// Configuration trait of this pallet.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Overarching event type
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		/// The base call filter to be used in normal operating mode
+		/// (When we aren't in the middle of a migration)
+		type NormalCallFilter: Filter<Self::Call>;
+		/// The base call filter to be used when we are in the middle of migrations
+		/// This should be very restrictive. Probably not allowing anything except possibly
+		/// something like sudo or other emergency processes
+		type MaintenanceCallFilter: Filter<Self::Call>;
+		/// The origin from which the call to enter or exit maintenance mode must come
+		/// Take care when choosing your maintenance call filter to ensure that you'll still be
+		/// able to return to normal mode. For example, if your MaintenanceOrigin is a council, make
+		/// sure that your councilors can still cast votes.
+		type MaintenanceOrigin: EnsureOrigin<Self::Origin>;
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// The chain was put into Maintenance Mode
+		EnteredMaintenanceMode,
+		/// The chain returned to its normal operating state
+		NormalOperationResumed,
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn maintenance_mode)]
+	/// Whether the site is in maintenance mode
+	type MaintenanceMode<T: Config> = StorageValue<_, bool, ValueQuery>;
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Place the chain in maintenance mode
+		///
+		/// Weight cost is two DB writes (1 for the mode and 1 for the event)
+		#[pallet::weight(2 * T::DbWeight::get().write)]
+		pub fn enter_maintenance_mode(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+			// Ensure Origin
+			T::MaintenanceOrigin::ensure_origin(origin)?;
+
+			// Write to storage
+			MaintenanceMode::<T>::put(true);
+
+			// Event
+			<Pallet<T>>::deposit_event(Event::EnteredMaintenanceMode);
+
+			Ok(().into())
+		}
+
+		/// Return the chain to normal operating mode
+		///
+		/// Weight cost is two DB writes (1 for the mode and 1 for the event)
+		#[pallet::weight(2 * T::DbWeight::get().write)]
+		pub fn resume_normal_operation(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+			// Ensure Origin
+			T::MaintenanceOrigin::ensure_origin(origin)?;
+
+			// Write to storage
+			MaintenanceMode::<T>::put(false);
+
+			// Event
+			<Pallet<T>>::deposit_event(Event::NormalOperationResumed);
+
+			Ok(().into())
+		}
+	}
+
+	impl<T: Config> Filter<T::Call> for Pallet<T> {
+		fn filter(call: &T::Call) -> bool {
+			if MaintenanceMode::<T>::get() {
+				T::NormalCallFilter::filter(call)
+			} else {
+				T::MaintenanceCallFilter::filter(call)
```

Comment:
```suggestion
			if MaintenanceMode::<T>::get() {
				T::MaintenanceCallFilter::filter(call)
			} else {
				T::NormalCallFilter::filter(call)
```
right? maintenance mode => maintenance call filter

could also do 
```suggestion
			if !MaintenanceMode::<T>::get() {
				T::NormalCallFilter::filter(call)
			} else {
				T::MaintenanceCallFilter::filter(call)
```

Reply:
Wow, that was a pretty significant bug. Thanks!